### PR TITLE
Make watchdog print 'short dumps' every 5 seconds the server is frozen

### PR DESCRIPTION
**Pull request by hafarooki**
This makes watchdog more useful by giving a more complete picture of what's freezing the main thread when it's frozen. It also helps finding the cause of short freezes without crashing the whole server whenever it freezes for 5 seconds.

### CODE REVIEW

File: Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+-                // Paper start - Warn in watchdog if an excessive velocity was ever set
+-                if ( org.bukkit.craftbukkit.CraftServer.excessiveVelEx != null )
+-                {
+-                    log.log( Level.SEVERE, "------------------------------" );
+-                    log.log( Level.SEVERE, "During the run of the server, a plugin set an excessive velocity on an entity" );
+-                    log.log( Level.SEVERE, "This may be the cause of the issue, or it may be entirely unrelated" );
+-                    log.log( Level.SEVERE, org.bukkit.craftbukkit.CraftServer.excessiveVelEx.getMessage());
+-                    for ( StackTraceElement stack : org.bukkit.craftbukkit.CraftServer.excessiveVelEx.getStackTrace() )
++                // Paper start - Different message when it's a short timeout
++                if(isLongTimeout) {
```
Comment (DenWav) [inline Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch]: Match the style of the surrounding code, please (even if said style is ugly, which it is). Keeping consistency is more important.

File: Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
++                    log.log( Level.SEVERE, "The server has stopped responding!" );
++                    log.log( Level.SEVERE, "Please report this to https://github.com/PaperMC/Paper/issues" );
++                    log.log( Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports" );
++                    log.log( Level.SEVERE, "Paper version: " + Bukkit.getServer().getVersion() );
++                    //
++                    if(net.minecraft.server.World.haveWeSilencedAPhysicsCrash)
```
Comment (DenWav) [inline Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch]: Same problem here.

File: Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
++                    {
++                        log.log( Level.SEVERE, "------------------------------" );
++                        log.log( Level.SEVERE, "During the run of the server, a physics stackoverflow was supressed" );
++                        log.log( Level.SEVERE, "near " + net.minecraft.server.World.blockLocation);
++                    }
++                    // Paper start - Warn in watchdog if an excessive velocity was ever set
++                    if ( org.bukkit.craftbukkit.CraftServer.excessiveVelEx != null )
```
Comment (DenWav) [inline Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch]: This is correct.

**Review by DenWav — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch (hunk 1 above)
Comment (kashike) [reply Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch]: In addition, do not indent any of this code - insert the `if` statement around the code without re-formatting.

**Pull request closed without merging**